### PR TITLE
add constexpr containers

### PR DESCRIPTION
**Pull request by schaumb**

### CODE REVIEW

File: include/magic_enum_containers.hpp [hunk 1]
```diff
@@ -0,0 +1,1361 @@
...
+  struct name_sort_impl {
+    [[nodiscard]] constexpr inline bool operator()(E e1, E e2) const noexcept {
+      return OP{}(enum_name(e1), enum_name(e2));
+    }
+  };
+
+  template<typename E,
+           typename OP>
+  struct name_sort_impl<E, OP, std::enable_if_t<!std::is_void_v<E> && is_transparent_v<OP>>> {
+    using is_transparent = std::true_type;
```
Comment (schaumb) [inline include/magic_enum_containers.hpp]: Fake, not transparent

File: include/magic_enum_containers.hpp [hunk 2]
```diff
@@ -0,0 +1,1361 @@
...
+                                            (std::is_same_v<Less, std::less<E>> || std::is_same_v<Less, std::less<>>)>> {
+    constexpr static inline const std::array<E, enum_count<E>()>* values_v = &enum_values<E>();
+    [[nodiscard]] constexpr inline optional<std::size_t> operator()(E val) const noexcept {
+      return enum_index(val);
+    }
+  };
+
+  template<>
+  struct indexing<void, std::less<>, void> {
+    using is_transparent = std::true_type;
```
Comment (schaumb) [inline include/magic_enum_containers.hpp]: It's only a noise, not needed.

File: include/magic_enum_containers.hpp [hunk 3]
```diff
@@ -0,0 +1,1361 @@
...
+      }
+    }
+  };
+
+  struct raw_access_t {};
+
+
+  template<typename Parent, typename Iterator, typename Getter, typename Predicate>
+  struct FilteredIterator {
+    Parent parent;
```
Comment (schaumb) [inline include/magic_enum_containers.hpp:232]: It is very heavy class, probably parent and current must be enough

**Comment by florianbecker** [comment]
@schaumb
So what's to do here? Why is this a draft PR? The last commits are 5 months ago.

I think we have to finish just one for the moment, and then we could discuss if we also want to add a magic_enum_map like #181.

**Comment by schaumb** [comment]
@florianbecker 
This PR is draft because it need to be add tests and documentation.

map: While I started to implement enum_map, I concluded [this](https://github.com/Neargye/magic_enum/issues/150#issuecomment-1136217197).

If you had time to add tests and documentation feel free to open another PR or PR to my fork's branch.

**Comment by florianbecker** [comment]
@schaumb 
Ok, I'll do so. Already working on it.

Please provide me with your use cases.

**Comment by schaumb** [comment]
@florianbecker made a lot of work on the tests and on the documentation. I mark this PR to ready for review.

**Review by Neargye — APPROVED** [review]

**Pull request merged**